enterprise/frontend/src/embedding-sdk/components/public/dashboard/SdkDashboard.tsx

```diff
@@ -257,12 +282,14 @@ const SdkDashboardInner = ({
           </SdkDashboardProvider>
         ))
         .with("queryBuilder", () => (
-          <InteractiveQuestionProvider questionId="new">
-            <InteractiveQuestionDefaultView
-              withResetButton
-              withChartTypeSelector
-            />
-          </InteractiveQuestionProvider>
+          <DashboardQueryBuilder
+            targetDashboardId={dashboardId}
+            onSave={(question) => {
+              setNewDashboardQuestionId(question.id);
+              setRenderMode("dashboard");
+              refetchDashboardRef.current();
```

Comment:
refetching the dashboard will also cause the dashcard to be refetched and that's how we will see the new dashcard in the UI.

By the way, the dashcard is already added to the dashboard when we click save on the save modal.